src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (this.health > 0) {
+      Animal.alive.push(this);
+    }
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+  bite(creature) {
+    if (creature instanceof Herbivore && creature.hidden !== true) {
+      creature.health -= 50;
+    }
+    Animal.alive = Animal.alive.filter(x => x.health > 0);
```

Comment:
Use better naming here
```suggestion
    Animal.alive = Animal.alive.filter(animal => animal.health > 0);
```

Reply:
After adding better naming move this to Animal class and put inside new method e.g. die and then reuse it here 
```suggestion
    animal.die();
```